jmolecules-ddd/src/main/java/org/jmolecules/ddd/annotation/Association.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jmolecules.ddd.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * An association to an {@link org.jmolecules.ddd.types.AggregateRoot}.
+ *
+ * @author Simon Zambrovski
+ * @see <a href="https://scabl.blogspot.com/2015/04/aeddd-9.html>John Sullivan - Advancing Enterprise DDD - Reinstating
+ * the Aggregate</a>
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE, ElementType.FIELD})
```

Comment:
Should we allow methods here as well. Should it be "annotation type" instead of "type"?

Reply:
An association could make sense on a getter, since we use Java, you are right:
METHOD, ANNOTATION TYPE, FIELD. Type makes actually no sense, since it is always a "usage" annotation and not a "definition" annotation... 

Agree will fix it...